operations/maintainer.go

```diff
@@ -0,0 +1,89 @@
+package operations
+
+import (
+	"context"
+	"github.com/Peripli/service-manager/pkg/log"
+	"github.com/Peripli/service-manager/pkg/query"
+	"github.com/Peripli/service-manager/pkg/types"
+	"github.com/Peripli/service-manager/pkg/util"
+	"github.com/Peripli/service-manager/storage"
+	"time"
+)
+
+// OperationMaintainer ensures that operations old enough are deleted
+// and that no stuck (orphan) operations are left in the DB due to crashes/restarts of SM
+type OperationMaintainer struct {
+	repository      storage.Repository
+	jobTimeout      time.Duration
+	cleanupInterval time.Duration
+}
+
+// NewOperationMaintainer constructs an OperationMaintainer
+func NewOperationMaintainer(repository storage.Repository, options *Settings) *OperationMaintainer {
+	return &OperationMaintainer{
+		repository:      repository,
+		jobTimeout:      options.JobTimeout,
+		cleanupInterval: options.CleanupInterval,
+	}
+}
+
+// Run starts the two recurring jobs responsible for cleaning up old operations
+// and deleting stuck (orphan) operations
+func (om *OperationMaintainer) Run() {
+	go om.cleanupOperations()
+	go om.cleanupStuckOperations()
+}
+
+// cleanOperations cleans up periodically all operations but the last
+// for each C/U/D operation for every resource_id which are older than some specified time
+func (om *OperationMaintainer) cleanupOperations() {
+	ticker := time.NewTicker(om.cleanupInterval)
+	terminate := make(chan struct{})
```

Comment:
who writes in this channel and terminates the maintainer. Probably it is better to have a context here and use it's done chan

Reply:
yep, will use the smCtx here